db/write_thread.h

```diff
@@ -401,6 +413,16 @@ class WriteThread {
   port::Mutex stall_mu_;
   port::CondVar stall_cv_;
 
+  // Count the number of stalls begun, so that we can check whether
+  // a particular stall has cleared (even if caught in another stall).
```

Comment:
I got a bit stuck on this at first thinking caught in another stall means multiple simultaneous stalls until remembering `BeginWriteStall()` cannot be called in that way. If it is indeed constrained like that maybe mention/assert that `stall_ended_count_ <= stall_begun_count_ <= stall_ended_count_ + 1`